docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
-
+## Design
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Good Class
+#### Description
+The goods class keeps track of the various inventory that will be input into the system through the commands class.
+The diagram below shows the model component of the orderline class.
+![Good Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Good.puml)
+
+The Goods class allows for the creation of orderline objects which has the following attributes: id, name, quantity and 
+description. Each attribute can be obtained using public get methods, and the attribute quantity can be set using the public set method.
+
+
+## Implementation
+### Regex Class
+#### Description
+The Regex Class is created to simplify the use of regular expressions in Java.
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Regex.puml)
```

Comment:
I think there might be some issues with the `dummyClass` calling `Regex`, as it should be a new activation bar (and the previous activation bar should have ended prior). 

![image](https://user-images.githubusercontent.com/79437795/160971678-295bc037-1c3a-4e3d-bdb2-a0de3cad7e76.png)